### PR TITLE
Annotate property change support

### PR DESCRIPTION
**Pull request by jpschewe**
Add nullness annotations for

- java.beans.BeanDescriptor
- java.beans.PropertyChangeSupport
- java.beans.Beans

### CODE REVIEW

**Comment by mernst** [comment]
Thank you for these annotations!  I apologize for the delay in reviewing the pull request.

I made a few tweaks to your annotations.  Can you double-check them?  Then I will merge this pull request.

Thanks again.  I appreciate your contribution.

File: src/java.desktop/share/classes/java/beans/PropertyChangeSupport.java [hunk 1]
```diff
@@ -353,7 +353,7 @@ public void addPropertyChangeListener(
         }
     }
 
-    private static void fire(@Nullable PropertyChangeListener[] listeners, PropertyChangeEvent event) {
+    private static void fire(PropertyChangeListener @Nullable [] listeners, PropertyChangeEvent event) {
```
Comment (jpschewe) [inline src/java.desktop/share/classes/java/beans/PropertyChangeSupport.java:356]: Why is @Nullable after "PropertyChangeListener"? The array of listeners can be null, so I would have through the type name is "@Nullable PropertyChangeListener".

Comment (mernst) [reply src/java.desktop/share/classes/java/beans/PropertyChangeSupport.java:356]: Thanks for the followup.

The reason is that a type annotation modifies the (part of the) type that immediately follows it.

`@Nullable PropertyChangeListener[] listeners` means "non-null array of possibly-null PropertyChangeListener".

`PropertyChangeListener @Nullable [] listeners` means "possibly-null array of non-null PropertyChangeListener"

This is also discussed at https://checkerframework.org/manual/#faq-array-syntax-meaning .

Comment (jpschewe) [reply src/java.desktop/share/classes/java/beans/PropertyChangeSupport.java:356]: Got it.

**Pull request merged**